### DISCUSSION TITLE
Translations for sentences with form inputs that form part of the sentence?

### DISCUSSION THREAD

**Discussion by angusclarkdev** [amannn/next-intl#935] [Q&A] [answered]
I'm currently building a feature that I'm struggling to think of how to provide translations for. Consider the following sentence where [] are inputs that the user can manipulate, where the initial values might be:

const energy = 'energy'
const percentage = "12%"
const months = 9

"the cost of [energy] will rise by [percentage] over the next [months] months"

How should I handle translations where the position of the inputs in the sentence will likely change depending on the language? We could break out 'the cost of', '*energy dropdown', 'will rise by" etc. But depending on the language, word order might be different, and I feel like doing it in that way would make for a really weird looking and hard to maintain messages json.

Any suggestions / experiences / patterns to leverage from the community would be much appreciated!

**Comment by rodrigovcortezi** [comment, marked as answer]
If I understand your problem correctly, you can use [interpolation of dynamic values](https://next-intl-docs.vercel.app/docs/usage/messages#interpolation-of-dynamic-values). Then for each language you can set the placeholder for these values in different position in the sentence.
Additionally, if you need the value itself to be translated, you can translate it before sending as a parameter in the interpolation. Example:
```javascript
const energy = t('energy')
const sentence = t('sentence', {energy})
```

**Reply by angusclarkdev** [reply]
I ended up doing something similar to this but I needed to pass components as dynamic values so made use of the t.rich() function. 

Thanks for the nudge in the right direction!
